base/applications/cleanmgr/dialog.c

```diff
@@ -102,26 +113,35 @@ INT_PTR CALLBACK StartDlgProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lPa
 
 INT_PTR CALLBACK ProgressDlgProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
 {   
-    switch(message)
+    switch (message)
     {
         case WM_INITDIALOG:
         {
+            HANDLE ThreadObj = NULL;
             WCHAR FullText[ARR_MAX_SIZE] = { 0 };
-            WCHAR SysDrive[ARR_MAX_SIZE] = { 0 };
             WCHAR TempText[ARR_MAX_SIZE] = { 0 };
-            HANDLE ThreadOBJ = NULL;
+            WCHAR *TempPtr = NULL;
 
             IsSystemDrive = FALSE;
             LoadStringW(GetModuleHandleW(NULL), IDS_SCAN, TempText, _countof(TempText));
-            StringCchPrintfW(FullText, sizeof(FullText), TempText, DriveLetter);
+            StringCchPrintfW(FullText, sizeof(FullText), TempText, SelectedDriveLetter);
             SetDlgItemTextW(hwnd, IDC_STATIC_SCAN, FullText);
-            GetEnvironmentVariableW(L"SystemDrive", SysDrive, sizeof(SysDrive));
-            if (wcscmp(SysDrive, DriveLetter) == 0)
+
+            /* In order to get the root drive, GetWindowsDirectoryW() function is getting
+               called and then the drive letter is extracted from the retrieved path */
+            GetWindowsDirectoryW(TempText, MAX_PATH);
+            TempPtr = wcschr(TempText, L'\\');
+            TempText[wcslen(TempText) - wcslen(TempPtr)] = L'\0';
```

Comment:
you know that these point to the same block of memory, since you just set one to point inside the other,
why not subtract pointers from eachother instead of calculating the string length on both?